### PR TITLE
[TSQL] Create columnstore indexes

### PR DESCRIPTION
**Pull request by paysni**
### Brief summary of the change made
This PR allows create columnstore index statements to be parsed.
https://learn.microsoft.com/en-us/sql/t-sql/statements/create-columnstore-index-transact-sql?view=sql-server-ver15
Fixes issue #5706 

### Are there any other side effects of this change that we should be aware of?
no side effects that I know of.

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_tsql.py [hunk 1]
```diff
@@ -1016,6 +1017,98 @@ class CreateIndexStatementSegment(BaseSegment):
...
 
 
+class CreateColumnstoreIndexStatementSegment(BaseSegment):
+    """A `CREATE COLUMNSTORE INDEX` statement.
+
+    https://learn.microsoft.com/en-us/sql/t-sql/statements/create-columnstore-index-transact-sql?view=sql-server-ver15
+    """
+
+    type = "create_columnstore_index_statement"
+    _on_partitions = Sequence(
```
Comment (greg-finley) [inline src/sqlfluff/dialects/dialect_tsql.py:1027]: The repo's code style is just putting these inline so `match_grammar` just reads top to bottom. You could put comments above certain sections if you wish.

**Review by greg-finley — COMMENTED** [review]
Great start! One small nit.

File: src/sqlfluff/dialects/dialect_tsql.py (hunk 1 above)
Comment (paysni) [reply src/sqlfluff/dialects/dialect_tsql.py]: Moved the _on_partitions and _with_option into match_grammar.

**Review by greg-finley — APPROVED** [review]
Thanks! 🎉

**Pull request merged**